src/System.Private.ServiceModel/tests/Common/Scenarios/Endpoints.cs

```diff
@@ -1,49 +1,183 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.ServiceModel;
+using System.Security.Cryptography.X509Certificates;
+
 using Infrastructure.Common;
 
 public static partial class Endpoints
 {
+    private static X509Certificate2 rootCert;
+    private static X509Certificate2 clientCert;
+    private static object rootCertLock = new object();
+    private static object clientCertLock = new object();
+    //Install Root certificate, this is required for all https test
+    private static void EnsureRootCertificateInstalled()
+    {
+        lock (rootCertLock)
+        {
+            if (rootCert == null)
+            {
```

Comment:
We call `EnsureRootCertificateInstalled()` in multiple places. 

If the root cert fails the first time, by https://github.com/dotnet/wcf/pull/1002/files#r59512010 bullet 1, it should fail every time we call this. Hence, we should use some kind of fail fast mechanism to fail all further calls to EnsureRootCertificateInstalled() in a particular run. 

We can't detect this condition currently because we are only keying off if `rootCert == null`